### ISSUE TITLE
password reset link robustness

### ISSUE THREAD

**Issue by haraldschilly** [sagemathinc/smc#294]
enhance the logic to handle password reset links in such a way, that it handles all uppercase letters. 

in particular, some MTAs convert the url to all uppercase and hence the anchor tag is `#FORGOT-1234-ABCD-...`

**Comment by williamstein** [comment]
Still an issue?

**Comment by williamstein** [comment]
If this is a problem it could be responsible for a lot of support issues!

**Comment by williamstein** [comment]
Put back the vague tag, as I still have no clue what this even means.

**Comment by haraldschilly** [comment]
I fixed some of the tags, e.g. no idea why this is backend. I didn't remember why I made the ticket, but now I digged up the motivation for it. It's based on zendesk's #666 and has to do with the MTA of a user, which did convert all characters to uppercase.

The fix might be to lowercase all anchor links in the webapp before processing them, or maybe just for the password reset case ...

More general, we could switch to doing password resets almost entirely on dedicated static websites without any javascript involvement. That helps to cover this case, since URLs aren't case sensitive.
